web/src/routes/PoisPage.tsx

```diff
@@ -55,30 +55,36 @@ const moveViewToBBox = (bBox: BBox, defaultVp: MapViewMercatorViewport): MapView
 }
 
 const PoisPage = ({ cityCode, languageCode, cityModel, pathname, languages }: CityRouteProps): ReactElement => {
-  const [queryParams, setQueryParams] = useSearchParams()
+  const { urlSlug } = useParams()
+  const navigate = useNavigate()
   const { data, error: featureLocationsError, loading } = useFeatureLocations(cityCode, languageCode)
   const [mapRef, setMapRef] = useState<Map | null>(null)
   const [snapPoint, setSnapPoint] = useState<number>(1)
-  const selectedFeatureSlug = queryParams.get(nameQueryParam)
   const [currentFeature, setCurrentFeature] = useState<PoiFeature | null>(
-    data?.features.find(it => it.properties.urlSlug === selectedFeatureSlug) ?? null
+    data?.features.find(it => it.properties.urlSlug === urlSlug) ?? null
   )
-  const poi = data?.pois.find(it => it.urlSlug === selectedFeatureSlug)
+  const poi = data?.pois.find(it => it.urlSlug === urlSlug)
   const { viewportSmall, height } = useWindowDimensions()
   const sheetRef = useRef<BottomSheetRef>(null)
   const [feedbackModalRating, setFeedbackModalRating] = useState<FeedbackRatingType | null>(null)
   const { t } = useTranslation('pois')
 
+  const getNavigationPath = (feature: PoiFeature | null, hasCurrentFeature: boolean): string => {
+    const pathname = window.location.pathname.replace(/\/$/, '')
+    if (feature) {
+      if (hasCurrentFeature) {
+        return `${getParentPath(pathname)}/${feature.properties.urlSlug}`
+      }
+      return `${pathname}/${feature.properties.urlSlug}`
+    }
+    return getParentPath(pathname)
+  }
+
   const selectFeature = (feature: PoiFeature | null) => {
     if (mapRef?.isMoving()) {
       mapRef.stop()
     }
-    if (feature) {
-      queryParams.set(nameQueryParam, feature.properties.urlSlug)
-    } else {
-      queryParams.delete(nameQueryParam)
-    }
-    setQueryParams(queryParams)
+    navigate(getNavigationPath(feature, !!currentFeature))
```

Comment:
```suggestion
    navigate(feature?.properties.urlSlug ?? '.')
```
Alternatively, if you want to do it explicitly, you should use the pathname functionality of the api-client to construct our pathnames.

Reply:
fixed in last commit.